coderd/healthcheck/healthcheck.go

```diff
@@ -36,6 +36,8 @@ type Report struct {
 	Time time.Time `json:"time"`
 	// Healthy is true if the report returns no errors.
 	Healthy bool `json:"healthy"`
+	// Warning is true when Coder is operational but its performance might be impacted.
+	Warning bool `json:"warning"`
```

Comment:
What if we get a report that has both `{ Healthy: false, Warning: true }`?
As far as I understand, that's an invalid state in our current model, which should not be representible.

Reply:
You're right, but I'm afraid that it will not be achievable without breaking API? I introduced the "warning" field to be perceived as "it is healthy, but...".

To solve it properly, most likely we need to replace "Healthy" with enum, for instance: `status: healthy`, `status: degraded`, `status: error`, etc.

I'm happy to chat about this as it is the right moment.

Reply:
The more I think about it, it could be useful to have both health and warn information, as they are not necessarily completely separate.

For example, if you're using embedded database and access URL just points straight to the VM, and you get a failed healthcheck for database and warnings for access URL taking too long to respond, that can already give you an indication that perhaps the server is overloaded.

Maybe we could add

```
Warnings []string or map[string]string
```

at the top level, and any warnings from component reports get added in?

Reply:
I agree with this concept if only we present these warnings on the "Health" page 👍 Otherwise, it is pointless. It also means that the Health page will become more informative than just Healthy/Unhealthy + JSON dump.